src/v/cluster/partition.h

```diff
@@ -208,6 +208,12 @@ class partition {
         return _archival_meta_stm;
     }
 
+    /// Return true if shadow indexing is enabled for the partition
+    bool is_remote_fetch_enabled() const {
+        const auto& cfg = _raft->log_config();
+        return cfg.is_shadow_indexing_fetch_enabled();
+    }
```

Comment:
nit: it still kind of bothers me that we have both "remote" and "shadow_indexing" names in the code, but I guess we can deal with it later.